lidder.go

```diff
@@ -184,7 +208,14 @@ func main() {
 		oops(err)
 	}
 
-	err = results.exploreDir(".")
+	singleFileMode := false
```

Comment:
I think it would be clearer to separate the single file mode completely. In the code here in `main`, you define `singleFileMode`, and have two places with `if/else` based on it. I'd rather have

```
var handler func() bool
if len(os.Args) == 2 {
  handler = fullScanMode
} else if len(os.Args) == 3 {
  handler = singleFileMode
} else {
  ... print usage and os.Exit(1)
}

if ok := handler(); !ok {
  os.Exit(1)
}
os.Exit(0)
```